addons/html_editor/controllers/main.py

```diff
@@ -592,8 +592,11 @@ def link_preview_metadata_internal(self, preview_url):
             Actions = request.env['ir.actions.actions']
             context = dict(request.env.context)
             words = preview_url.strip('/').split('/')
-
-            record_id = int(words.pop())
+            # If there is no recordID in url return empty dict.
```

Comment:
it feels like we should not have to change this. I think that a better solution would be to change the proper js code to not call this route when there is no record id in the preview_url.